editor/src/clj/editor/code/script.clj

```diff
@@ -510,11 +514,20 @@
             (preprocessors/preprocess-lua-lines lua-preprocessors lines resource :debug)
             (catch Exception exception
               exception))]
-      (if (ex-message preprocessed-lines)
+      (if-some [exception-message (ex-message preprocessed-lines)]
         (let [exception preprocessed-lines
-              message (format "Lua preprocessing failed for file '%s'. Check the editor logs for exception info." (resource/proj-path resource))]
-          (log/error :message message :exception exception)
-          (g/->error _node-id :build-targets :fatal resource message))
+              build-error-message (str "Lua preprocessing failed.\n" exception-message)
+              log-error-message (format "Lua preprocessing failed for file '%s'." (resource/proj-path resource))]
+          (log/error :message log-error-message :exception exception)
+          (if-some [[proj-path line-number] (try-parse-file-line exception-message)]
+            (let [project (project/get-project _node-id)
+                  exception-resource (workspace/resolve-resource resource proj-path)
+                  exception-node-id (project/get-resource-node project exception-resource)
+                  error-node-id (or exception-node-id _node-id)
+                  error-resource (if (nil? exception-node-id) resource exception-resource)
+                  error-user-data (r/make-code-error-user-data proj-path line-number)]
+              (g/->error error-node-id :modified-lines :fatal error-resource build-error-message error-user-data))
+            (g/->error _node-id :modified-lines :fatal resource build-error-message)))
```

Comment:
Previously we were reporting the error on the `:build-targets` output, but it really is the `:modified-lines` property that is the source of these errors, so I changed it.

I don't think the label is used for anything in the error - it's mostly a mechanism for us to diagnose problems when developing.